### PR TITLE
add keyEncryption with Xor implementation

### PR DESCRIPTION
**Pull request by sboishtyan**
Feature add api for encryption keys. That allow hide original keys name on the disk

### CODE REVIEW

File: library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    private String xorName(String name) {
+        byte[] beforeXor = name.getBytes();
+        int beforeXorLength = beforeXor.length;
+        int xorLength = xor.length;
+        byte[] afterXor = new byte[beforeXorLength];
+        for (int beforeXorIndex = 0; beforeXorIndex < beforeXorLength; beforeXorIndex++) {
+            byte afterXorItem = beforeXor[beforeXorIndex];
+            for (int xorIndex = 0; xorIndex < xorLength; xorIndex++) {
```
Comment (iamironz) [inline library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java]: I think inner loop should be as an separate method. Also this method should take raw byte and return xor'ed byte. In other words inner loop and inner loop indexes code parts should be fully incapsulated.

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * @author Sergey Boishtyan
+ */
+public class XorKeyEncryptionImplTest {
+
+    private final XorKeyEncryptionImpl xorKeyEncryption = new XorKeyEncryptionImpl("111111111111111011".getBytes());
+
+    @Test
+    public void decrypt() throws Exception {
```
Comment (iamironz) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: checked exceptions will never throws

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        String origin = "origin";
+        String decrypt = xorKeyEncryption.decrypt(origin);
+        assertNotEquals(origin, decrypt);
+
+        String encrypt = xorKeyEncryption.encrypt(decrypt);
+        assertEquals(origin, encrypt);
+    }
+
+    @Test
+    public void encrypt() throws Exception {
```
Comment (iamironz) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: same as decrypt

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * @author Sergey Boishtyan
+ */
+public class XorKeyEncryptionImplTest {
```
Comment (iamironz) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: Also you should check that xor'ed byte array will be incorrect de-xor'ed with another xor byte array. Just watch on `AesByteEncryptionImpl.java`.

Comment (iamironz) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: Also you should check `checkMirror` method with even size, non-even size with non-mirrored byte array and the same with mirrored byte array.

Comment (iamironz) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: Plus check minimum key length.

File: library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                afterXorItem ^= xor[xorIndex];
+            }
+            afterXor[beforeXorIndex] = afterXorItem;
+        }
+
+        return new String(afterXor);
+    }
+
+    private void checkMirror() {
+        if (isEven()) {
```
Comment (iamironz) [inline library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java]: Invert if condition for decreased code nesting.

File: library/src/main/java/com/ironz/binaryprefs/encryption/KeyEncryption.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.ironz.binaryprefs.encryption;
+
+/**
+ * @author Sergey Boishtyan
```
Comment (sboishtyan) [inline library/src/main/java/com/ironz/binaryprefs/encryption/KeyEncryption.java]: done

File: library/src/main/java/com/ironz/binaryprefs/encryption/KeyEncryption.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+public interface KeyEncryption {
```
Comment (sboishtyan) [inline library/src/main/java/com/ironz/binaryprefs/encryption/KeyEncryption.java]: done

File: library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java (hunk 1 above)
Comment (sboishtyan) [inline library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java]: done

File: library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java (hunk 5 above)
Comment (sboishtyan) [inline library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java]: Code nesting isn't so big, and revert condition like `if(isOdd){ return;} else {//...}` i don't like. May be we can stay it like this?

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java (hunk 4 above)
Comment (sboishtyan) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: done

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java (hunk 2 above)
Comment (sboishtyan) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: done

File: library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java (hunk 3 above)
Comment (sboishtyan) [inline library/src/test/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImplTest.java]: done

File: library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java (hunk 5 above)
Comment (iamironz) [inline library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java]: Ok

**Pull request merged**